src/EnergyPlus/AirflowNetworkBalanceManager.cc

```diff
@@ -8984,6 +8989,31 @@ namespace AirflowNetworkBalanceManager {
                 }
             }
 
+            // Check number of fans specified in an AirLoop #6748
+            for (BranchNum = 1; BranchNum <= PrimaryAirSystem(1).NumBranches; ++BranchNum) {
+                NumOfFans = 0;
+                FanNames = "";
+                for (CompNum = 1; CompNum <= PrimaryAirSystem(1).Branch(BranchNum).TotalComponents; ++CompNum) {
+                    if (UtilityRoutines::SameString(PrimaryAirSystem(1).Branch(BranchNum).Comp(CompNum).TypeOf, "Fan:ConstantVolume") ||
+                        UtilityRoutines::SameString(PrimaryAirSystem(1).Branch(BranchNum).Comp(CompNum).TypeOf, "Fan:OnOff") ||
+                        UtilityRoutines::SameString(PrimaryAirSystem(1).Branch(BranchNum).Comp(CompNum).TypeOf, "Fan:VariableVolume")) {
+                        NumOfFans++;
+                        if (NumOfFans > 1) {
+                            FanNames += PrimaryAirSystem(1).Branch(BranchNum).Comp(CompNum).Name;
+                            break;
+                        } else {
+                            FanNames += PrimaryAirSystem(1).Branch(BranchNum).Comp(CompNum).Name + ",";
+                        }
+                    }
+                }
+                if (NumOfFans > 1) break;
+            }
+            if (NumOfFans > 1) {
+                ShowSevereError(RoutineName + "An AirLoop branch, " + PrimaryAirSystem(1).Branch(BranchNum).Name + ", has two or more fans: " + FanNames);
+                ShowContinueError("The AirflowNetwork model allows a single supply fan in an AirLoop only. Please make changes in the input file accordingly.");
+                ErrorsFound = true;
+            }
```

Comment:
I think we could put this in the loop above and avoid this extra section down here.

Reply:
@Myoldmopar The loop above is tested based on AirflowNetwork nodes and links. However, the user file shows a supply fan defined AirflowNetwork:Distribution:Component:Fan and a return fan defined in AirflowNetwork:Distribution:Component:ReliefAirFlow. In reality, a return fan is a component in an AirLoop in the user input. It is an unexpected configuration. That is why I used an extra section to issue a fatal error based on PrimaryAirSystem information.